workshops/views.py

```diff
@@ -1017,14 +1046,50 @@ def event_delete(request, event_ident):
 def event_import(request):
     """Read tags from remote URL and return them as JSON.
 
-    This is used to read tags from workshop index page and then fill up fields
+    This is used to read tags from workshop website and then fill up fields
     on event_create form."""
+
+    url = request.POST['url'].strip()
     try:
-        url = request.POST['url']
-        translated_data = parse_tags_from_event_index(url)
-        return JsonResponse(translated_data)
-    except (KeyError, WrongEventURL):
-        raise SuspiciousOperation('Missing or wrong `url` POST parameter.')
+        # fetch page
+        response = requests.get(url)
+        response.raise_for_status()  # assert it's 200 OK
+        content = response.text
+
+        # find tags
+        tags = find_tags_on_event_website(content)
+
+        if 'slug' not in tags:
+            # there are no HTML tags, so let's try the old method
+            url = generate_url_to_event_index(url)
+
+            # fetch page
+            response = requests.get(url)
+
+            if response.status_code == 200:
+                # don't throw errors for pages we fall back to
+                content = response.text
+                tags = find_tags_on_event_index(content)
+
+        # normalize (parse) them
+        tags = parse_tags_from_event_website(tags)
+
+        # TODO: change JavaScript for handling import/update from URL
+
+        return JsonResponse(tags)
+
+    except requests.exceptions.HTTPError as e:
+        raise SuspiciousOperation(
+            'Request for "{0}" returned status code {1}.'
+            .format(url, e.response.status_code)
+        )
+
+    except (requests.exceptions.ConnectionError,
+            requests.exceptions.Timeout):
+        raise SuspiciousOperation('Network connection error.')
+
+    except KeyError:
+        raise SuspiciousOperation('Missing or wrong "url" POST parameter.')
```

Comment:
There's a lot of code between the `try` and this `except`, so I'm concerned that you'll get false catches of *other* `KeyError`s here.  In general, it's good practice to use very explicit exceptions (e.g. `requests.exceptions.ConnectionError`) and/or make the code covered by a try/except block as small as possible.